torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -505,13 +505,18 @@ def forward(self, *inputs: Any) -> Any:
                 aug = self._apply_image_or_video_transform(
                     aug, transform_id, magnitude, interpolation=self.interpolation, fill=self.fill
                 )
-            mix.add_(
-                # The multiplication below could become in-place provided `aug is not batch and aug.is_floating_point()`
-                # Currently we can't do this because `aug` has to be `unint8` to support ops like `equalize`.
-                # TODO: change this once all ops in `F` support floats. https://github.com/pytorch/vision/issues/6840
-                combined_weights[:, i].reshape(batch_dims)
-                * aug
-            )
+            weights = combined_weights[:, i].reshape(batch_dims)
+            # We use the is operator to cheaply check if the `aug` reference we have is the same as the `batch`.
+            # This can happen if we repeatedly sample the identity operator. Because all operators avoid in-place
+            # modifications, this is a cheap and safe way to detect it.
+            if aug is batch or not aug.is_floating_point():
+                # When the `aug` is not variable is not floating point or if it's the same as the original batch
+                # we can't do in-place operations.
+                aug = aug.mul(weights)
+            else:
+                # We can do in-place on all other cases.
+                aug.mul_(weights)
```

Comment:
Turns out this is not beneficial and just complicates the implementation. See benchmark:
```
[-------------- AutoAugment cpu torch.float32 --------------]
                         |        old       |        new     
1 threads: --------------------------------------------------
      (3, 256, 256)      |    7 (+-  4) ms  |    6 (+-  0) ms
      (16, 3, 256, 256)  |  119 (+- 74) ms  |  120 (+- 72) ms
6 threads: --------------------------------------------------
      (3, 256, 256)      |    8 (+-  1) ms  |    8 (+-  1) ms
      (16, 3, 256, 256)  |  114 (+- 77) ms  |  117 (+- 78) ms

Times are in milliseconds (ms).

[-------------- AutoAugment cuda torch.float32 -------------]
                         |        old       |        new     
1 threads: --------------------------------------------------
      (3, 256, 256)      |    2 (+-  0) ms  |    2 (+-  0) ms
      (16, 3, 256, 256)  |    2 (+-  0) ms  |    2 (+-  0) ms
6 threads: --------------------------------------------------
      (3, 256, 256)      |    2 (+-  0) ms  |    2 (+-  0) ms
      (16, 3, 256, 256)  |    2 (+-  0) ms  |    2 (+-  0) ms

Times are in milliseconds (ms).

[--------------- AutoAugment cpu torch.uint8 ---------------]
                         |        old       |        new     
1 threads: --------------------------------------------------
      (3, 256, 256)      |    8 (+-  1) ms  |    8 (+-  1) ms
      (16, 3, 256, 256)  |  154 (+- 73) ms  |  145 (+- 78) ms
6 threads: --------------------------------------------------
      (3, 256, 256)      |   10 (+-  1) ms  |   10 (+-  1) ms
      (16, 3, 256, 256)  |  177 (+- 82) ms  |  179 (+- 90) ms

Times are in milliseconds (ms).

[--------------- AutoAugment cuda torch.uint8 --------------]
                         |        old       |        new     
1 threads: --------------------------------------------------
      (3, 256, 256)      |    2 (+-  0) ms  |    2 (+-  0) ms
      (16, 3, 256, 256)  |    2 (+-  0) ms  |    2 (+-  0) ms
6 threads: --------------------------------------------------
      (3, 256, 256)      |    2 (+-  0) ms  |    2 (+-  0) ms
      (16, 3, 256, 256)  |    2 (+-  0) ms  |    2 (+-  0) ms

Times are in milliseconds (ms).

[----------------- AutoAugment cpu pil -----------------]
                     |        old       |        new     
1 threads: ----------------------------------------------
      (3, 256, 256)  |    9 (+-  1) ms  |    9 (+-  1) ms
6 threads: ----------------------------------------------
      (3, 256, 256)  |   11 (+-  1) ms  |   11 (+-  1) ms

Times are in milliseconds (ms).
```

I think we should just remove the TODO.